doc/cascadia/Command-Palette-spec.md

```diff
@@ -0,0 +1,269 @@
+heavily lean on the "focused" element to determine which terminal is "active".
+However, when the command plaette is opened, focus will move out of the terminal
+control into the command palette, which leads to some hard to debug crashes.
+
+## Future considerations
+
+* Once [#1142] is also complete, we'll also add a `executeCommand` action. It
+  will take a single parameter `command`, which will be the name of a command
+  from the list of commands. This will enable a user to bind a command to a
+  keybinding. This could allow the user to bind a specific pairing of `action`
+  and `args` that they've set up in a command as a keybinding, without having to
+  repeat the entry in both the `commands` and `keybindings`.
+* Commands will provide an easy point for allowing an extension to add it's
+  actions to the UI, without forcing the user to bind the extension's actions to
+  a keybinding
+* Also discussed in [#2046] was the potential for adding a command that inputs a
+  certain commandline to be run by the shell. I felt that was out of scope for
+  this spec, so I'm not including it in detail. I believe that would be
+  accomplished by adding a `inputCommand` action, with two args: `commandline`,
+  a string, and `suppressNewline`, an optional bool, defaulted to false. The
+  `inputCommand` action would deliver the given `commandline` as input to the
+  connection, followed by a newline (as to execute the command).
+  `suppressNewline` would prevent the newline from being added. This would work
+  relatively well, so long as you're sitting at a shell prompt. If you were in
+  an application like `vim`, this might be handy for executing a sequence of
+  vim-specific keybindings. Otherwise, you're just going to end up writing a
+  commadline to the buffer of vim. It would be weird, but not unexpected.
+* Additionally mentioned in [#2046] was the potential for profile-scoped
+  commands. While that's a great idea, I believe it's out of scope for this
+  spec.
+* Once [#754] lands, we'll need to make sure to include commands for each action
+  manually in the default settings. This will add some overhead that the
+  developer will need to do whenever they add an action. That's unfortunate, but
+  will be largely beneficial to the end user.
+* We could theoretically also display the keybinding for a certain command in
+  the `ListViewItem` for the command. We'd need some way to correlate a
+  command's action to a keybinding's action. This could be done in a follow-up
+  task.
+* We might want to alter the fuzzy-search algorithim, to give higher precedence
+  in the results list to commands with more consecutive matching characters.
+  Alternatively we could give more weight to commands where the search matched
+  the initial character of words in the command.
+  - For example: `ot` would give more weight to "**O**pen **T**ab" than
+    "**O**pen Se**t**tings").
+* Another idea for a future spec is the concept of "nested commands", where a

```

Comment:
Note to future me: maybe we want these nested commands to be evaluated when the parent is opened, like "open new tab..." then queries all the profiles to _dynamically_ create entries.